codersdk/agentsdk/logs.go

```diff
@@ -89,164 +83,3 @@ func LogsWriter(ctx context.Context, sender func(ctx context.Context, log ...Log
 		source: source,
 	}
 }
-
-// LogsSenderFlushTimeout changes the default flush timeout (250ms),
-// this is mostly useful for tests.
-func LogsSenderFlushTimeout(timeout time.Duration) func(*logsSenderOptions) {
-	return func(o *logsSenderOptions) {
-		o.flushTimeout = timeout
-	}
-}
-
-type logsSenderOptions struct {
-	flushTimeout time.Duration
-}
-
-// LogsSender will send agent startup logs to the server. Calls to
-// sendLog are non-blocking and will return an error if flushAndClose
-// has been called. Calling sendLog concurrently is not supported. If
-// the context passed to flushAndClose is canceled, any remaining logs
-// will be discarded.
-func LogsSender(sourceID uuid.UUID, patchLogs func(ctx context.Context, req PatchLogs) error, logger slog.Logger, opts ...func(*logsSenderOptions)) (sendLog func(ctx context.Context, log ...Log) error, flushAndClose func(context.Context) error) {
```

Comment:
There are external consumers of this SDK, e.g. https://github.com/coder/envbuilder/blob/eb069e462b1db798fe2bde4628296c796938e113/cmd/envbuilder/main.go#L56

Should we perhaps consider moving the enqueue portion into here so that it can be more readily utilized e.g. by envbuilder?

Reply:
Yeah, what I'll do is leave these alone in this PR, then create a PR that moves `logSender` here (and exports it and relevant methods).  That PR will mark this as deprecated.